### PR TITLE
Add TypeScript definition

### LINKED ISSUE #7: Missing typescript declarations
Granted, this is a very minor issue.

### PR DESCRIPTION
**Pull request by PopGoesTheWza**
Fixes #7

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+import normalizeNewline from 'normalize-newline';
+
+normalizeNewline('foo\r\nbar\nbaz');
+//=> 'foo\nbar\nbaz'
+
+normalizeNewline(Buffer.from('foo\r\nbar\nbaz')).toString();
+//=> 'foo\nbar\nbaz'
+```
+*/
+export function normalizeNewline(input: string): string;
```
Comment (sindresorhus) [inline index.d.ts]: It's a default export

File: index.d.ts [hunk 2]
```diff
@@ -11,5 +11,6 @@ normalizeNewline(Buffer.from('foo\r\nbar\nbaz')).toString();
 //=> 'foo\nbar\nbaz'
 ```
 */
-export function normalizeNewline(input: string): string;
-export function normalizeNewline(input: Buffer): Buffer;
+declare function normalizeNewline(input: string): string;
+declare function normalizeNewline(input: Buffer): Buffer;
+export default normalizeNewline;
```
Comment (sindresorhus) [inline index.d.ts]: Still incorrect. No need for `declare`.

Comment (PopGoesTheWza) [reply index.d.ts]: Hum... but with no 'declare' I get `Top-level declarations in .d.ts files must start with either a 'declare' or 'export' modifier.`

So if I am incorrect, I am likely clueless too :/

Comment (PopGoesTheWza) [reply index.d.ts]: You don't fancy this one-liner, do you?
```ts
export default function normalizeNewline<Input extends string | Buffer>(input: Input): Input;
```

**Comment by sindresorhus** [comment]
It should be:

```ts
export default function normalizeNewline(input: string): string;
export default function normalizeNewline(input: Buffer): Buffer;
```

You might need to upgrade XO to allow that.

**Comment by PopGoesTheWza** [comment]
> You might need to upgrade XO to allow that.

`xo` now screams at me for reasons I fail to grasp:

```sh
  index.test-d.ts:4:1
  ✖  4:1  Unsafe call of an any typed value.  @typescript-eslint/no-unsafe-call
  ✖  5:1  Unsafe call of an any typed value.  @typescript-eslint/no-unsafe-call
```

**Comment by sindresorhus** [comment]
I fixed that issue elsewhere now.

**Comment by PopGoesTheWza** [comment]
> I fixed that issue elsewhere now.

Uh? It still occurs on my side... but you're the boss. :)

**Comment by sindresorhus** [comment]
Tests are passing

**Pull request merged**